File: Android/doraemonkit/src/main/java/com/didichuxing/doraemonkit/kit/dbdebug/DbDebugKit.java
package com.didichuxing.doraemonkit.kit.dbdebug;

import android.content.Context;
import android.content.Intent;
import android.os.Build;
import android.widget.TextView;

import com.amitshekhar.DebugDB;
import com.amitshekhar.debug.encrypt.sqlite.DebugDBEncryptFactory;
import com.amitshekhar.debug.sqlite.DebugDBFactory;
import com.didichuxing.doraemonkit.DoraemonKit;
import com.didichuxing.doraemonkit.R;
import com.didichuxing.doraemonkit.constant.BundleKey;
import com.didichuxing.doraemonkit.constant.FragmentIndex;
import com.didichuxing.doraemonkit.kit.Category;
import com.didichuxing.doraemonkit.kit.IKit;
import com.didichuxing.doraemonkit.ui.TranslucentActivity;
import com.didichuxing.doraemonkit.ui.UniversalActivity;
import com.didichuxing.doraemonkit.util.netstate.NetType;
import com.didichuxing.doraemonkit.util.netstate.NetWork;
import com.didichuxing.doraemonkit.util.netstate.NetworkManager;

/**
 * ================================================
 * 作    者：jint（金台）
 * 版    本：1.0
 * 创建日期：2019-09-24-17:05
 * 描    述：数据库远程访问入口
 * 修订历史：
 * ================================================
 */
public class DbDebugKit implements IKit {
    @Override
    public int getCategory() {
        return Category.TOOLS;
    }

    @Override
    public int getName() {
        return R.string.dk_tools_dbdebug;
    }

    @Override
    public int getIcon() {
        return R.drawable.dk_db_view;
    }

    @Override
    public void onClick(Context context) {
        Intent intent = new Intent(context, UniversalActivity.class);
        intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
        intent.putExtra(BundleKey.FRAGMENT_INDEX, FragmentIndex.FRAGMENT_DB_DEBUG);
        context.startActivity(intent);

    }

    @Override
    public void onAppInit(Context context) {
        NetworkManager.init(context);
        NetworkManager.get().register(this);
        DebugDB.initialize(DoraemonKit.APPLICATION, new DebugDBFactory());
        DebugDB.initialize(DoraemonKit.APPLICATION, new DebugDBEncryptFactory());
    }

    @NetWork(NetType.WIFI)
    public void listenWifi(NetType type) {
        if (type == NetType.NONE) {
            DebugDB.shutDown();
        } else {
            if (!DebugDB.isServerRunning()) {
                DebugDB.initialize(DoraemonKit.APPLICATION, new DebugDBFactory());
                DebugDB.initialize(DoraemonKit.APPLICATION, new DebugDBEncryptFactory());
            }
        }
    }
}

File: Android/dokit/src/main/java/com/didichuxing/doraemonkit/kit/dbdebug/DbDebugKit.kt
package com.didichuxing.doraemonkit.kit.dbdebug

import android.app.Activity
import android.content.Context
import com.didichuxing.doraemonkit.R
import com.didichuxing.doraemonkit.kit.AbstractKit

/**
 * ================================================
 * 作    者：jint（金台）
 * 版    本：1.0
 * 创建日期：2019-09-24-17:05
 * 描    述：数据库远程访问入口 去掉
 * 修订历史：
 * ================================================
 */
//@AutoService(AbstractKit.class)
class DbDebugKit : AbstractKit() {
    override val name: Int
        get() = R.string.dk_tools_dbdebug
    override val icon: Int
        get() = R.mipmap.dk_db_view

    override fun onClickWithReturn(activity: Activity): Boolean {
        return true
    }

    override fun onAppInit(context: Context?) {}
    override val isInnerKit: Boolean
        get() = true

    override fun innerKitId(): String {
        return "dokit_sdk_comm_ck_dbview"
    }
}